app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -1,28 +1,33 @@
 class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
-  # You should configure your model like this:
-  # devise :omniauthable, omniauth_providers: [:twitter]
+  def facebook
+    callback_from(:facebook)
+  end
 
-  # You should also create an action method in this controller like this:
-  # def twitter
-  # end
+  def passthru
+    super
+  end
 
-  # More info at:
-  # https://github.com/plataformatec/devise#omniauth
+  def failure
+    redirect_to root_path
+  end
 
-  # GET|POST /resource/auth/twitter
-  # def passthru
-  #   super
-  # end
+  protected
 
-  # GET|POST /users/auth/twitter/callback
-  # def failure
-  #   super
-  # end
+  def after_omniauth_failure_path_for(scope)
+    super(scope)
+  end
 
-  # protected
+  private
 
-  # The path used when OmniAuth fails
-  # def after_omniauth_failure_path_for(scope)
-  #   super(scope)
-  # end
+  def callback_from(provider)
+    provider = provider.to_s
```

Comment:
__[rubocop]__
* Useless assignment to variable - `provider`. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

<!-- Comment by [SideCI](https://sideci.com) -->